homeassistant/components/modbus/light.py

```diff
@@ -0,0 +1,69 @@
+"""Support for Modbus lights."""
+import logging
+from typing import Any, Dict, Optional
+
+from homeassistant.components.light import LightEntity
+from homeassistant.const import CONF_LIGHTS, CONF_NAME
+from homeassistant.helpers.typing import (
+    ConfigType,
+    DiscoveryInfoType,
+    HomeAssistantType,
+)
+
+from .const import CALL_TYPE_COIL, CONF_INPUT_TYPE, MODBUS_DOMAIN
+from .modbus import ModbusHub
+from .switch import ModbusCoilSwitch, ModbusRegisterSwitch
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    async_add_entities,
+    discovery_info: Optional[DiscoveryInfoType] = None,
+):
+    """Read configuration and create Modbus light."""
+    if discovery_info is None:
+        return
```

Comment:
I wonder if it would be better to add a warning, so the user sees something is wrong. I fell into this a couple of times, and wondered what happened until I started the debugger. WDYT ?

Reply:
Sure, it will do no harm.

Reply:
I added an error message. Now it should be clear why our component wasn't initialized.